### PR TITLE
WW-5391 Add interface for VelocityManager extension point

### PR DESCRIPTION
**Pull request by kusalk**
WW-5391
--
Allows applications to provide a completely custom VelocityManager

### CODE REVIEW

File: plugins/osgi/src/main/java/org/apache/struts2/osgi/OsgiConfigurationProvider.java [hunk 1]
```diff
@@ -255,16 +254,27 @@ public void setBundleAccessor(BundleAccessor acc) {
     }
 
     @Inject
-    public void setVelocityManager(VelocityManager vm) {
-        LOG.trace("OSGi ConfigurationProvider - setVelocityManager() called - VelocityManager: [{}]", vm);
-
+    public void setVelocityManager(VelocityManagerInterface vmi) {
+        LOG.trace("OSGi ConfigurationProvider - setVelocityManager() called - VelocityManager: [{}]", vmi);
+        if (!(vmi instanceof VelocityManager)) {
```
Comment (kusalk) [inline plugins/osgi/src/main/java/org/apache/struts2/osgi/OsgiConfigurationProvider.java:259]: Didn't bother with a marker interface here as this plugin is deprecated anyway

File: plugins/portlet/src/main/java/org/apache/struts2/portlet/result/PortletVelocityResult.java
Comment (kusalk) [inline plugins/portlet/src/main/java/org/apache/struts2/portlet/result/PortletVelocityResult.java:1]: Standardised line endings to `LF`

File: plugins/velocity/src/main/java/org/apache/struts2/views/velocity/VelocityManagerInterface.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import org.apache.velocity.context.Context;
+
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * @since 6.4.0
+ */
+public interface VelocityManagerInterface {
```
Comment (kusalk) [inline plugins/velocity/src/main/java/org/apache/struts2/views/velocity/VelocityManagerInterface.java:32]: Maybe in 7.0 we can consider renaming `VelocityManager` to `StrutsVelocityManager` and `VelocityManagerInterface` to `VelocityManager`

**Review by lukaszlenart — COMMENTED** [review]
Looks good, are you going to address some of the reported issues by Sonar?

**Comment by kusalk** [comment]
@lukaszlenart Sonar results don't make any sense, it's been misconfigured ever since the 7.0 branch was introduced

**Comment by lukaszlenart** [comment]
Should be fixed

**Comment by kusalk** [comment]
@lukaszlenart Thanks! Looks like it's just complaining about deprecations, serialisation and code coverage. I've fixed the serialisation warnings. The code coverage I don't think it makes sense for me to rectify as I've just done a find and replace on injection methods. Adding coverage to those injection methods alone won't add much value, and it's out of scope for me to add coverage to the whole class.

**Review by lukaszlenart — APPROVED** [review]

**Pull request merged**